common/scala/src/main/scala/whisk/common/TransactionId.scala

```diff
@@ -200,58 +196,42 @@ object TransactionId {
+  val testing = TransactionId("sid_testing") // Common id for for unit testing
+  val invoker = TransactionId("sid_invoker") // Invoker startup/shutdown or GC activity
+  val invokerWarmup = TransactionId("sid_invokerWarmup") // Invoker warmup thread that makes stem-cell containers
+  val invokerNanny = TransactionId("sid_invokerNanny") // Invoker nanny thread
+  val dispatcher = TransactionId("sid_dispatcher") // Kafka message dispatcher
+  val loadbalancer = TransactionId("sid_loadbalancer") // Loadbalancer thread
+  val invokerHealth = TransactionId("sid_invokerHealth") // Invoker supervision
+  val controller = TransactionId("sid_controller") // Controller startup
+  val dbBatcher = TransactionId("sid_dbBatcher") // Database batcher
+
+  def apply(tid: String, extraLogging: Boolean = false): TransactionId = {
+    val now = Instant.now(Clock.systemUTC())
+    TransactionId(TransactionMetadata(tid, now, extraLogging))
   }
 
   implicit val serdes = new RootJsonFormat[TransactionId] {
     def write(t: TransactionId) = {
       if (t.meta.extraLogging)
-        JsArray(JsNumber(t.meta.id), JsNumber(t.meta.start.toEpochMilli), JsBoolean(t.meta.extraLogging))
+        JsArray(JsString(t.meta.id), JsNumber(t.meta.start.toEpochMilli), JsBoolean(t.meta.extraLogging))
       else
-        JsArray(JsNumber(t.meta.id), JsNumber(t.meta.start.toEpochMilli))
+        JsArray(JsString(t.meta.id), JsNumber(t.meta.start.toEpochMilli))
     }
 
     def read(value: JsValue) =
       Try {
         value match {
-          case JsArray(Vector(JsNumber(id), JsNumber(start))) =>
-            TransactionId(TransactionMetadata(id.longValue, Instant.ofEpochMilli(start.longValue), false))
-          case JsArray(Vector(JsNumber(id), JsNumber(start), JsBoolean(extraLogging))) =>
-            TransactionId(TransactionMetadata(id.longValue, Instant.ofEpochMilli(start.longValue), extraLogging))
+          case JsArray(Vector(JsString(id), JsNumber(start))) =>
+            TransactionId(TransactionMetadata(id, Instant.ofEpochMilli(start.longValue), false))
+          case JsArray(Vector(JsString(id), JsNumber(start), JsBoolean(extraLogging))) =>
+            TransactionId(TransactionMetadata(id, Instant.ofEpochMilli(start.longValue), extraLogging))
         }
       } getOrElse unknown
   }
 }
 
-/**
- * A thread-safe transaction counter.
- */
-trait TransactionCounter {
-  case class TransactionCounterConfig(stride: Int)
-
-  val transCounterConfig = loadConfigOrThrow[TransactionCounterConfig](ConfigKeys.transactions)
-  val stride = transCounterConfig.stride
-  val instanceOrdinal: Int
-
-  // seed the counter so transids do not overlap: instanceOrdinal + n * stride, start at n = 1
-  private lazy val cnt = new AtomicInteger(instanceOrdinal + stride)
-
-  def transid(extraLogging: Boolean = false): TransactionId = {
-    TransactionId(cnt.addAndGet(stride), extraLogging)
-  }
-}
+case class TransactionGeneratorConfig(header: String)
```

Comment:
I think for local testing it might still be nice to have a counter based id instead. Thoughts?

Reply:
I've addad a counter to the test. This might be helpful.